### PR TITLE
Leaks

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/p2p/net_node.inl [hunk 1]
```diff
@@ -714,6 +714,14 @@ namespace nodetool
   template<class t_payload_net_handler>
   bool node_server<t_payload_net_handler>::send_stop_signal()
   {
+    std::list<boost::uuids::uuid> connection_ids;
+    m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt) {
+      connection_ids.push_back(cntxt.m_connection_id);
+      return true;
+    });
+    for (const auto &connection_id: connection_ids)
+      m_net_server.get_config_object().close(connection_id);
```
Comment (vtnerd) [inline src/p2p/net_node.inl]: In addition to this work, should a destructor somewhere also clean up in case close was not called?

Comment (moneromooo-monero) [inline src/p2p/net_node.inl]: There is already one. Though it may all be buggy for all we know.

File: src/p2p/net_node.inl [hunk 2]
```diff
@@ -714,6 +714,14 @@ namespace nodetool
   template<class t_payload_net_handler>
   bool node_server<t_payload_net_handler>::send_stop_signal()
   {
+    std::list<boost::uuids::uuid> connection_ids;
+    m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt) {
```
Comment (vtnerd) [inline src/p2p/net_node.inl]: I didn't comment on this last time, is it possible to close the connection in the `foreach_connection` callback? Or does that wreck the config object somehow?

Comment (moneromooo-monero) [inline src/p2p/net_node.inl]: It might deadlock.

**Pull request merged**